pages/es6_example/home.page.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * Created by titusfortner on 11/23/16.
+ */
+const Page = require('./page');
+
+class HomePage extends Page {
+
+    get a_link() { return browser.element("a[id='i am a link']"); };
+    get input_comment() { return browser.element('#comments'); };
+    get submit_form() { return browser.element('#submit'); };
+    get comment_text() { return browser.element('#your_comments'); } 
+
+    open() {
+       super.open("training-test-page");
+    }
+
+    setComment(comment) {
+        this.input_comment.setValue(comment);
+    }
+
+    follow_link() {
+        this.a_link.click()
+    }
+
+    submit() {
+        this.submit_form.click();
+        let el = this.comment_text;
+
+        browser.waitUntil(function () {
```

Comment:
I didn't see any browser being instantiated, how are you just using it? Is it part of some module?

Reply:
It's sort of global (closure scoped, I think), but dependent on the runner.  To make it foolproof, you could do
```
constructor(browser) {
  this.browser = browser;
}
```
but then you'd need to use `this.browser` everywhere.  Since the test runner allows it and it's the convention, I think it's ok.

Reply:
WebdriverIO handles browser instantiation and quitting internally. By executing `npm test` in the root of this project, this will execute the WebdriverIO runner. This allows us to use a `browser` object in tests and page objects (or elsewhere) without explicitly creating a new instance. This approach is a bit different from other language frameworks/Webdriver bindings but is typical for Javascript tools like WebdriverIO or Protractor.

Reply:
I think we should leave it as is and not have a `browser` argument passed to the page object constructor. The short explanation for this "we don't have to, so we won't". 

The longer explanation is that using these classes is syntactic sugar in NodeJS, and actually aren't "classes" in the Java/C# sense of the term. These are JavaScript functions under the hood and most things have global scope (such as `browser`). Members (`this.browser` for example) might be closures and have some kind of non-global scope but my gut tells me that the `browser` instance is global anyway (which is why we can access it everywhere). This means we can have simpler or even default constructors and don't have to pass around a driver instance to our page objects. Since we're not using actual class-based object oriented programming, we don't have to adhere to that approach and we can use a more "JS-like" approach.